tpm/internal/key/key.go

```diff
@@ -108,6 +108,20 @@ type CreateConfig struct {
 	Size int
 }
 
+func (c *CreateConfig) Validate() error {
+	switch c.Algorithm {
+	case "RSA":
+		if c.Size > 2048 {
+			return fmt.Errorf("%d bits RSA keys are (currently) not supported; maximum is 2048", c.Size)
+		}
```

Comment:
Is this strictly true? Is it because of go-attestation, or is it a limitation of the TPMs we have used?

Reply:
I've added `go.step.sm/crypto` explicitly to the error message in faa54c3e06ca7e2dfca6fd7af81abfb5f1a0e585.

Could also prefix it, but `crypto` is less descriptive as to where it comes from by just reading the message.

The limitation is with the TPMs we've been testing with. At least that's my verdict right now. I see this error message as something temporary, until we add support for 1) checking the error resulting from the TPM and translating that to something more user friendly (ideally in a lower level, such as `go-tpm`) and/or 2) check the actual parameter values a specific TPM (model) allows before performing the operation (e.g. similar to the `tpm2_testparms` utility; and in a way that we can be sure that we're right about the capabilities of the TPM).